### PR TITLE
Test functional tests

### PR DESCRIPTION
**Pull request by ATmobica**
#### Problem
Mbed CHIP integration functional testing support

 #### Summary of Changes
Add functional tests for shell example and lock-app based on Pytest - src/test_driver/mbed-functional dir
Changed setup_network_env script to setup environment for functional testing
More details about functional testing:  [Mbed CHIP functional testing](https://github.com/ARMmbed/connectedhomeip/wiki/Mbed-CHIP-functional-testing)

### CODE REVIEW

File: src/test_driver/mbed-functional/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,49 @@
+appdirs==1.4.4
```
Comment (pan-) [inline src/test_driver/mbed-functional/requirements.txt:1]: We will have to add this into the pigweed requirements when upstreaming.

File: src/test_driver/mbed-functional/shell/test_server.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    ret = connected_device.wait_for_output("Echo Message")
+    assert ret != None
+
+    # Disable server
+    ret = connected_device.send(command="server off", expected_output="Done", wait_before_read=5)
+    assert ret != None and len(ret) > 1
+
+# Not supported
+# @pytest.mark.networktest
+# def test_server_tcp(connected_device, chip_tools_dir):
```
Comment (pan-) [inline src/test_driver/mbed-functional/shell/test_server.py:42]: What's missing for this test ?

**Review by pan- — APPROVED** [review]
When the fixes are merged, I merge that one too.

File: src/test_driver/mbed-functional/requirements.txt (hunk 1 above)
Comment (ATmobica) [reply src/test_driver/mbed-functional/requirements.txt:1]: Okay

File: src/test_driver/mbed-functional/shell/test_server.py (hunk 2 above)
Comment (ATmobica) [reply src/test_driver/mbed-functional/shell/test_server.py:42]: Generally, it is an issue to accept a new connection on TCP server so the test failed. However, I left it to continue debugging. TCP server doesn't seem to be the major issue for now and we decided to wait with it to upstream reabase.

**Pull request merged**